### ISSUE TITLE
Output of ADAS features

### ISSUE THREAD

**Issue by ashwinsarvesh** [ApolloAuto/apollo#11799]
Hello,

I want to use the output of ADAS features like lane detection, traffic light detection, and obstacle detection to compare it with the output of my ADAS camera and evaluate the performance of my camera. 

Is it possible to extract the output of Apollo 3.0 ADAS features on a PC? In what format can the data be obtained?

**Comment by michael4338** [comment]
merging from [11081](https://github.com/ApolloAuto/apollo/issues/11801) Apollo 3.0 have the features Lane departure warning, forward collision warning, and Traffic sign recognition?

**Comment by ashwinsarvesh** [comment]
Any help anyone?

**Comment by jeroldchen** [comment]
Hi @ashwinsarvesh , what kind of output are you looking for exactly? The detection boxes?

**Comment by ashwinsarvesh** [comment]
I need the output of Lane detection and obstacle detection to compare with the same of an ADAS camera. This is to compare the deviation in the performance of Apollo 3.0 and my ADAS camera.

**Comment by ashwinsarvesh** [comment]
(ROS topic /apollo/perception/obstacles)  
Is this the ROS topic which I need to subscribe to get the output of the perception module? (Lane detection + Obstacle detection)

**Comment by jeroldchen** [comment]
@ashwinsarvesh  Yes, it is. For the data structure of PerceptionObstacles, please refer to [https://github.com/ApolloAuto/apollo/blob/r3.0.0/modules/perception/proto/perception_obstacle.proto](url) .

**Comment by ashwinsarvesh** [comment]
Thank you :)
